### DISCUSSION TITLE
Relaxed density matrix for TDDFT excited states

### DISCUSSION THREAD

**Discussion by tomfay** [pyscf/pyscf#2649] [Q&A] [answered]
Hi - is there a way currently to obtain the relaxed density matrix and/or Z matrix for TD-DFT excited states? I think I've found the relevant lines for obtaining Z from the TDRKS gradients (https://github.com/pyscf/pyscf/blob/master/pyscf/grad/tdrks.py lines 116-127) but I was wondering if there's a more straightforward way to obtain the relaxed density matrix. Thanks!

**Comment by sunqm** [comment, marked as answer]
I'm not sure whether there is a widely accepted definition of "relaxed density matrix" in the post-HF methods. Therefore, I never implement an API to provide such density matrices, although you might find some functions in the package called respons_density_matrix or relaxed_density_matrix. If the Z vector is considered as the occ-virt block of the relaxed density matrix, as you found in the TDRKS implementation, you can extract the `dm1` to define your density matrix from https://github.com/pyscf/pyscf/blob/df093594f27af8cebe546f08a38277f3e40bd63f/pyscf/grad/tdrks.py#L142 .

**Comment by verena-neufeld** [comment]
Hi @tomfay , thanks for your question. Has it been resolved and can this issue be closed?
